tests/test_dataloader.py

```diff
@@ -339,3 +391,85 @@ def test_finetuning_dataloader_small_data(dataset_size: int,
 
     if dist.get_global_rank() == 0:
         shutil.rmtree(tiny_dataset_folder_path)
+
+
+@pytest.mark.parametrize('add_bad_data_dropped', [True, False])
+@pytest.mark.parametrize('add_bad_data_error', [True, False])
+def test_malformed_data(
+    add_bad_data_dropped: bool,
+    add_bad_data_error: bool,
+):
+    tokenizer_name = 'mosaicml/mpt-7b'
+    max_seq_len = 2048
+    dataset_size = 5
+    device_batch_size = 5
+    tiny_dataset_folder_path = os.path.join(os.getcwd(), 'test-ift-data-small')
```

Comment:
nit: maybe use tmp_path fixture if possible here, so you don't need to worry about removing the folder

Reply:
yeah makes sense, not sure why I did it the other way.